dcc-portal-ui/app/scripts/advanced/js/advanced.js

```diff
@@ -63,10 +63,24 @@ angular.module('icgc.advanced.controllers', [
           _serviceMap = {},
           _filterService = LocationService.getFilterService();
 
+      _controller.SSMDonorCount = 0;
+
       _locationFilterCache = _filterService.getCachedFiltersFactory();
 
       var _isInAdvancedSearchCtrl = true;
 
+      _controller.fetchSSMDonorCount = (filters) => {
+        filters = LocationService.merge(filters, {'donor':{'availableDataTypes':{'is':['ssm']}}});
+        AdvancedDonorService.getSSMDonorCount(filters)
+          .then(function (count) {
+            _controller.SSMDonorCount = count;
+          });
+      }
+
```

Comment:
We should now be able to use some async/await sugar and make our control flow a bit easier to follow

See if this works?
```js
_controller.fetchSSMDonorCount = async (filters) => {
  filters = LocationService.merge(filters, {'donor':{'availableDataTypes':{'is':['ssm']}}});
  _controller.SSMDonorCount = await AdvancedDonorService.getSSMDonorCount(filters);
}
```

Good article on async/await:

https://ponyfoo.com/articles/understanding-javascript-async-await